protocols/gossipsub/src/handler.rs

```diff
@@ -77,6 +77,9 @@ pub enum HandlerIn {
 /// creation loops.
 const MAX_SUBSTREAM_ATTEMPTS: usize = 5;
 
+/// The maximum number of rpc messages in the send queue before inbound processing is put on hold.
+const SEND_QUEUE_CAPACITY: usize = 16;
+
```

Comment:
With https://github.com/libp2p/rust-libp2p/pull/4778/commits/c5d4df0e23a2d452d9308843f217910c7f3784e5, this change can be removed, right?

A note on the general idea of not reading when the send queue is full. I think this only has an impact if one has a single gossip peer only. If one has more than one gossip peer, one might get more inbound messages through other `ConnectionHandler`s (i.e. connections) even if this `ConnectionHandler` is not reading.

Reply:
> With https://github.com/libp2p/rust-libp2p/commit/c5d4df0e23a2d452d9308843f217910c7f3784e5, this change can be removed, right?

That's right. I just pushed a commit addressing that.

> A note on the general idea of not reading when the send queue is full. I think this only has an impact if one has a single gossip peer only. If one has more than one gossip peer, one might get more inbound messages through other ConnectionHandlers (i.e. connections) even if this ConnectionHandler is not reading.

Right, I didn't think of this. In that case, it looks like rate-limiting from the behaviour itself is the better option